Mods can specify required Highlander version in config

Iridar opened the pull request:
Closes #909

Localized descriptions are placeholder and we should probably have a discussion about them. Once we finalize them, I can get a person to translate them (and other localized Highlander text) to Russian. 

I didn't commentate the code because it's relatively straightforward and heavily based on Musashi's implementation of #524.

robojumper commented on X2WOTCCommunityHighlander/Src/X2WOTCCommunityHighlander/Classes/X2WOTCCH_ModDependencies.uc:
```diff
@@ -174,4 +182,41 @@ function static string Join(array<string> StringArray, optional string Delimiter
+{
+    return CurrentCHLVersion.MajorVersion < RequiredCHLVersion.MajorVersion ||
+           CurrentCHLVersion.MinorVersion < RequiredCHLVersion.MinorVersion ||
+           CurrentCHLVersion.PatchVersion < RequiredCHLVersion.PatchVersion;
```
I'm afraid this is not correct. If current were `1.22.0` and required is at least `1.21.1`, then this would say the HL is outdated.

Iridar replied:
Good eye, I'll get that fixed.

Iridar commented:
I've added a separate commit that makes `DisplayName` in `CHModDependency` a `config` var. It appears this was an oversight (bug) in the initial implementation. The comment mentions this string being "equivalent to empty if not specified in localization", but the commented out code example assigns `DisplayName` in config, right along other config variables. This is also where I've been assigning "DisplayName" for all of my mods that used this feature while being unaware of this bug. So I've adjusted the comment as well.

robojumper commented:
Status: This will almost certainly be on the receiving end of a merge conflict due to #966, which I want to merge first (after Musashi gives it a quick look). Assigning waiting-on-author because of that.

pledbrook commented:
It seems unlikely that Musashi will be reviewing that DLC order PR any time soon, so should this PR really wait on that review? Perhaps we need to review how we're going to deal with the mod dependency and DLC run order PRs.

Iridar commented:
Replaced by #1007.
